straxen/plugins/peaklet_processing.py

```diff
@@ -207,6 +225,141 @@ def clip_peaklet_times(peaklets, start, end):
                 p['length'] = (end - p['time']) // p['dt']
 
 
+@numba.jit(nopython=True, nogil=True, cache=True)
+def peak_saturation_correction(records, peaks, to_pe,
+                               reference_length=100,
+                               min_reference_length=20,
+                               use_classification=False,
+                               ):
+    """Correct the area and per pmt area of peaks from saturation
+    :param records: Records
+    :param peaks: Peaklets / Peaks
+    :param reference_length: Maximum number of reference sample used
+    to correct saturated samples
+    :param min_reference_length: Minimum number of reference sample used
+    to correct saturated samples
+    :param use_classification: Choice to use peak type
+    """
+
+    if not len(records):
+        return
+    if not len(peaks):
+        return
+
+    # Search for peaks with saturated channels
+    mask = peaks['n_saturated_channels'] > 0
+    if use_classification:
+        mask &= peaks['type'] == 2
+    peak_list = np.where(mask)[0]
```

Comment:
FYI, this is interesting first I was thinking why you are not directly using something like `peaks[mask]` instead of your `peaks[np.where(mask)[0]]`. It turned out that the latter is about 10 % faster.

Reply:
Interesting insight, my intention was just for that deprecated return at the end. But 10% faster is a bonus?